src/cargo/util/toml/targets.rs

```diff
@@ -752,7 +752,7 @@ fn infer_any(package_root: &Path, entry: &DirEntry) -> Option<(String, PathBuf)>
 
 fn infer_file(package_root: &Path, entry: &DirEntry) -> Option<(String, PathBuf)> {
     let path = entry.path();
-    let stem = path.file_stem()?.to_str()?.to_owned();
+    let stem = path.file_name()?.to_str()?.to_owned();
```

Comment:
This seems to be the root cause which strips the extension from the bench name. However with this change it produces: "[ERROR] invalid character '.' in crate name: `benches.rs`" and I can't find where that error is produced :(

Reply:
That's probably the smart thing to do, for which I was eyeballing [add_extension](https://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html#method.add_extension). I don't suppose we can use this nightly API?

Reply:
No, cargo must work on stable.